sdk/iot_hub/src/service/requests/create_or_update_configuration.rs

```diff
@@ -0,0 +1,127 @@
+use azure_core::headers;
+use azure_core::Method;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::convert::TryInto;
+
+use crate::service::resources::{ConfigurationContent, ConfigurationMetrics};
+use crate::service::responses::ConfigurationResponse;
+use crate::service::{ServiceClient, API_VERSION};
+
+azure_core::operation! {
+    /// The CreateOrUpdateConfigurationBuilder is used to construct a new configuration
+    /// or update an existing one.
+    CreateOrUpdateConfiguration,
+    client: ServiceClient,
+    configuration_id: String,
+    priority: u64,
+    target_condition: String,
+    etag: Option<String>,
```

Comment:
Instead of exposing two operations, we could do what the Python SDK does, which is to have a single `create_or_update` function, and provide the optional parameter `if_match`.

https://docs.microsoft.com/en-us/python/api/azure-mgmt-iothub/azure.mgmt.iothub.v2021_07_02.operations.iothubresourceoperations?view=azure-python#azure-mgmt-iothub-v2021-07-02-operations-iothubresourceoperations-begin-create-or-update

> if_match
> ETag of the IoT Hub. Do not specify for creating a brand new IoT Hub. Required to update an existing IoT Hub.

Reply:
This could be done across the crate.

Reply:
That sounds fine. I think the entire client interface needs to be rethought through[^1] so maybe we leave that for a different PR?

[^1] This SDK does not follow the "client tree" model and instead ops for one mega client. We should probably move away from that.